### PR TITLE
[Feature]: Table Sorting, Filtering, Pagination

### PR DESCRIPTION
**Pull request by OustanDing**
- Added sorting, filtering and pagination to `Table` component
- Dependencies:
  - React Table
  - React Paginate (for pagination component)
- New components:
  - `Pagination` - Page selector for table pagination
  - `Pill` - Pill for filter option

_Note: In the future, we should rework the Performances Page filtering state. Right now, every possible filter has its own state, which requires a bunch of complex object destructuring logic and extra code._

### CODE REVIEW

File: pages/performances/index.js [hunk 1]
```diff
@@ -18,62 +22,302 @@ import styles from '@styles/pages/Performances.module.scss'; // Page styles
...
+  const [performanceLevelOptions, setPerformanceLevelOptions] = useState({});
+  const [danceStyleOptions, setDanceStyleOptions] = useState({});
+  const [danceSizeOptions, setDanceSizeOptions] = useState({});
+  const [filters, setFilters] = useState([]);
+  const [pageNumber, setPageNumber] = useState(0);
+  const [pageCount, setPageCount] = useState(0);
+
+  // Get initial filter options
+  useEffect(() => {
+    const getSettings = async () => {
```
Comment (mkanoria) [inline pages/performances/index.js:55]: Kinda confused here - is this supposed to be in the `page/performances/index.js` file?

File: pages/performances/index.js [hunk 2]
```diff
@@ -18,62 +22,302 @@ import styles from '@styles/pages/Performances.module.scss'; // Page styles
 // Temp constants
 import data, { columns } from '../../data/mockParticipants';
 
+const PAGE_SIZE = 3; // Rows per page
```
Comment (mkanoria) [inline pages/performances/index.js:25]: I'm assuming this changed based on which page we're on? Maybe this was copied over wrong from the setting page?

File: pages/performances/index.js [hunk 3]
```diff
@@ -84,13 +328,59 @@ export default function Performances() {
 }
 
 // Entries Table
-const EntryTable = () => {
-  return <Table columns={columns} data={data} />;
+const EntryTable = props => {
+  const columns = [
```
Comment (mkanoria) [inline pages/performances/index.js:332]: Is this different from the `columns` imported on line 23?

**Review by mkanoria — COMMENTED** [review]
Just a few small comments, but LGTM otherwise!!

File: pages/performances/index.js (hunk 1 above)
Comment (OustanDing) [reply pages/performances/index.js:55]: Yes, sorry about the confusing naming. Setting (especially school, dance style and dance level) are used to filter the performances, so they need to be retrieved to populate the filter dropdowns.

File: pages/performances/index.js (hunk 2 above)
Comment (OustanDing) [reply pages/performances/index.js:25]: This is a constant that sets the maximum number of rows per page. I set the value to 3 just for testing, but we can set it to whatever is needed.

File: pages/performances/index.js (hunk 3 above)
Comment (OustanDing) [reply pages/performances/index.js:332]: Yeah. The `columns` on line 23 is still the mock data which is currently used to populate the Judging table. The `columns` here on line 332 is for the Admin view table. Once the Judging table is done, we won't need the mock columns anymore.

File: pages/performances/index.js (hunk 1 above)
Comment (mkanoria) [reply pages/performances/index.js:55]: Ahh so to get the different options for each category, right? Sounds good then

File: pages/performances/index.js (hunk 3 above)
Comment (mkanoria) [reply pages/performances/index.js:332]: Right, that makes sense then. We can prob eventually move these column definitions into a separate file in the future

**Comment by mkanoria** [comment]
Just a clarification - this PR covers the new Table functionality + adding that to the Performance page, right? We can do something similar for settings now, I'm assuming?

**Comment by OustanDing** [comment]
> Just a clarification - this PR covers the new Table functionality + adding that to the Performance page, right? We can do something similar for settings now, I'm assuming?

@mkanoria Yeah, exactly. However, IIRC Julia said that we won't be having sorting/filtering/pagination for the settings page?

**Comment by mkanoria** [comment]
> > Just a clarification - this PR covers the new Table functionality + adding that to the Performance page, right? We can do something similar for settings now, I'm assuming?
> 
> @mkanoria Yeah, exactly. However, IIRC Julia said that we won't be having sorting/filtering/pagination for the settings page?

Ohh perfect, let's get this merged in then 💪🏽 🥳

**Review by thepenguinco — APPROVED** [review]
LGTM.  Let's see if we can get the data seeding in this cycle as well to test out the whole workflow.

**Review by mkanoria — APPROVED** [review]

**Pull request merged**